streams/src/main/java/org/apache/kafka/streams/processor/internals/PartitionGroup.java

```diff
@@ -93,6 +96,29 @@ long partitionTimestamp(final TopicPartition partition) {
         return queue.partitionTime();
     }
 
+    // creates queues for new partitions, removes old queues, saves cached records for previously assigned partitions
+    void updatePartitions(final Set<TopicPartition> newInputPartitions, final Function<TopicPartition, RecordQueue> recordQueueCreator) {
+        final Set<TopicPartition> removedPartitions = new HashSet<>();
+        final Iterator<Map.Entry<TopicPartition, RecordQueue>> queuesIterator = partitionQueues.entrySet().iterator();
+        while (queuesIterator.hasNext()) {
+            final Map.Entry<TopicPartition, RecordQueue> queueEntry = queuesIterator.next();
+            final TopicPartition topicPartition = queueEntry.getKey();
+            if (newInputPartitions.contains(topicPartition)) {
```

Comment:
remove should be a no-op if the topicPartition doesn't exist. Might be able to rephrase:
```
if (!newInputPartitions.contains(topicPartition)) {
   totalBuffered -= queueEntry.getValue().size();
   queuesIterator.remove();
   removedPartitions.add(topicPartition);
}
newInputPartitions.remove(topicPartition);

Reply:
Yes, can rephrased as you offer.